synapse/events/snapshot.py

```diff
@@ -198,6 +201,11 @@ def deserialize(storage: "Storage", input: JsonDict) -> "EventContext":
             rejected=input["rejected"],
         )
```

Comment:
> It looks like `_current_state_ids` is initialized to an [empty dict for outlier events](https://github.com/matrix-org/synapse/blob/v1.47.0rc2/synapse/events/snapshot.py#L134-L140), but that field is sent as a state group ID instead of the raw dict when the `EventContext` is serialized/deserialized
>
> *-- https://github.com/matrix-org/synapse/pull/11220/#discussion_r747057523*

As an alternative, would it be worth changing [`_AsyncEventContextImpl._fill_out_state`](https://github.com/matrix-org/synapse/blob/c99da2d0794de7240e791eba1a2f2d703fdecb60/synapse/events/snapshot.py#L320-L341) to default `current_state_ids={}` and `prev_state_ids={}` if there is no `state_group`?

Is there ever a case where `current_state_ids`/`prev_state_ids` should be `None` on purpose?

Reply:
Now it does that and it seems to work still